### PR TITLE
qemu: disable pdpe1gb on Intel Mac

### PR DESCRIPTION
**Pull request by AkihiroSuda**
https://stackoverflow.com/a/72863744/5167443

Workaround for #1485

### CODE REVIEW

**Review by jandubois — APPROVED** [review]
I haven't tested it, but patch looks good.

I'm just confused why this is not required with earlier versions of qemu; don't they enable large page sizes by default?

The SO entry seems to describe a different error/symptom; just the workaround seems to be the same.

But if prevents the crash, then this should be good.

File: pkg/limayaml/defaults.go [hunk 1]
```diff
@@ -120,6 +120,15 @@ func FillDefault(y, d, o *LimaYAML, filePath string) {
...
 			}
 		}
+		if arch == X8664 && runtime.GOOS == "darwin" {
+			switch cpuType[arch] {
+			case "host", "max":
+				// Disable pdpe1gb on Intel Mac
+				// https://github.com/lima-vm/lima/issues/1485
+				// https://stackoverflow.com/a/72863744/5167443
+				cpuType[arch] += ",-pdpe1gb"
+			}
```
Comment (jandubois) [inline pkg/limayaml/defaults.go:130]: I can't make up my mind if this should be mentioned in `defaults.yaml`:

```yaml
cpuType:
  # 🟢 Builtin default: "qemu64" (or "host" when running on x86_64 host)
  x86_64: null
```

Maybe the runtime warning is good enough? idk

Comment (AkihiroSuda) [reply pkg/limayaml/defaults.go:130]: Added

**Comment by AkihiroSuda** [comment]
> I'm just confused why this is not required with earlier versions of qemu; don't they enable large page sizes by default?

QEMU v8.0.0-rc1 bumped up the EDK2 binary (https://gitlab.com/qemu-project/qemu/-/commit/86305e864191123dcf87c3af639fddfc59352ac6), that might be the reason.

**Review by jandubois — APPROVED** [review]

**Pull request merged**